src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -589,16 +589,35 @@ private static int GetOpenBraceSpanEnd(SyntaxNode root)
         }
 
         private static int GetBraceInsertionPosition(SyntaxNode node)
-            => node switch
+        {
+            if (node is SwitchStatementSyntax switchStatementNode)
+            {
+                if (switchStatementNode.OpenParenToken.IsMissing && switchStatementNode.CloseParenToken.IsMissing)
+                {
+                    // There is no parenthesis pair in the switchStatementNode,
+                    // e.g.
+                    // void Foo(int i)
+                    // {
+                    //    var c = i swit$$ch
+                    // }
+                    // Consider this as a SwitchExpression, add the brace after 'switch'
+                    return switchStatementNode.SwitchKeyword.Span.End;
+                }
+
+                // In all other case, think it is a switch statement, add brace after the clos parenthesis.
+                return switchStatementNode.CloseParenToken.Span.End;
```

Comment:
Won't this break if a user has `switch (` and then presses Shift+Enter?

Reply:
No that won't happen because when 'CloseParenToken' is missing, it will be marked at the same position as the '(', so everything should still work except the final format might be strange. (Because the formatter can't do correct things)
Let me find if I can add a test here

Reply:
OK, test added